### DISCUSSION TITLE
Validation on the general shape of an uuid string

### DISCUSSION THREAD

**Discussion by TessavWalstijn** [uuidjs/uuid#747] [Ideas]
```javascript
import { validate } from 'uuid'

if (validate(randomUuid)) {
  // do stuff when uuid is valid
}
```
This is lovely however when a system or software product had an introduction of invalid id's this can be quite annoying.

The WHY:
Working with old data sets that have gotten a generated id that looks like an uuid but is not a valid one. The data set is to large and has to many links to each other that creating new valid uuid's is a no go. I currently use a custom RegEx check and would be happy to continue using that.

My proposal is to add a `softValidation`, I know this is not covered in the [RFC4122](https://www.ietf.org/rfc/rfc4122.txt).
And if it is by that reason not wanted to implement a `softValidation` I fully understand it. However having a somewhat a closer standard implemented by the uuidjs group would be nice to have a soft validation. 

Naming of the function is not set in stone.

How it will work:
```javascript
// soft-validate.js
const REGEX = /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i

function softValidate(id) {
  return typeof id === 'string' && REGEX.test(id);
}

export default softValidate;
```
It would match with the following ID's
```javascript
00000000-0000-0000-0000-000000000000 // null
b99e3058-ff1b-11ee-9ae6-325096b39f47 // v1
000003e8-ff1b-21ee-af00-325096b39f47 // v2
8fab4bca-2ade-38c8-b23a-9454a10a7fcd // v3
1bb2fffb-b24d-4bf3-9f2e-660ad7e697da // v4
9db6f67c-dd95-5ea0-aa5b-e70e5c5f7cf5 // v5
018efbb7-3e87-7b4d-a108-a1149c69e73a // v7
9bc1766a-c527-bd51-3f7d-d28eef5aab14 // invalid but looks on first glance OK
71d4868d-f9b1-73db-9825-32ab1547dbeb // invalid but looks on first glance OK
```

Notice:
- that in the 3rd block there is no version restriction
- that in the 4th block there is no 89AB restriction
- that it is still restricted to [0-9a-f] keeping it base 16
- yes softValidate is not [RFC4122](https://www.ietf.org/rfc/rfc4122.txt) standard

Note:
I would have no problem doing the work for this addition. But first want to check with the maintainers before I do the work and create a PR that will be closed anyway.

PS I have seen https://github.com/uuidjs/uuid/pull/72#discussion_r6502478 

Cheers!

**Comment by broofa** [comment]
You're welcome to implement this in a separate module, but it's not suitable for `uuid`.  This module is strictly about RFC4122-compliant UUIDs.
